### ISSUE TITLE
Cannot justify items centered

### ISSUE THREAD

**Issue by garretto** [obartra/reflex#240]
Trying to make a full-screen component with centered children.

```javascript
<Layout height="parent">
  <Root align="center" justify="center">
    <G>Want centered but this is left aligned</G>
  </Root>
</Layout>
```

![screen shot 2017-08-30 at 1 31 03 pm](https://user-images.githubusercontent.com/784852/29894691-1a422f9a-8d8b-11e7-916a-1617ff8100dd.png)

**Comment by arahansen** [comment]
I think you want `size="fit"` on the `<G>` element

Example: https://codesandbox.io/s/93l8jvzl8r

**Comment by garretto** [comment]
adding `justify="center"` also works but I was hoping to avoid adding an attribute to every child `G`

**Comment by obartra** [comment]
@arahansen setting size to `fit` would break the layout if we want each Component in a new line.

I think `<Grid>` should inherit the `justify` value of its parent. We can probably do this with just css. Right now we are not setting `text-align`. Maybe we should?

**Comment by obartra** [comment]
Thinking about it a bit more, this is tricky because for instance, if you want to right align a button you don't necessarily want the text to be right aligned (and you wouldn't expect that alignment to change just by changing the position of the button).

That being said, you could argue that if a Button should be centered that property will already be set in the button component overriding the parent default.

What are your thoughts? Another approach would be to add another parameter but that doesn't sound like a good approach either (at that point we may as well use css)

**Comment by obartra** [comment]
After further discussion with @arahansen we won't be tackling this issue. I don't think there would be much benefit from doing something like `textAlign="center"` versus using React's `style={{ textAlign: 'center' }}`.

We cannot assume either than a center aligned block will require center align text, or that a right aligned block will have right aligned text. They are independent properties. For instance, a right aligned button may have centered text.

@arahansen has put together a demo on a way to achieve this layout with reflex alone but it requires an additional `<Grid />` element. [See here](https://codesandbox.io/s/0p854329wv). If that's an ok tradeoff it may be a more convenient alternative.

**Comment by garretto** [comment]
Not going to lie, this drives me insane. Ran into this problem again today trying to right align children.  It's a common use case for me to have a column of items each on their own line that needs to be aligned in some way. 

I don't necessarily want items on in a new grid, just on a new line. Can we have a `flex-direction: column` kind of attribute?

https://codesandbox.io/s/5wz84l428n

**Comment by garretto** [comment]
This is my solution for now:

```javascript
import React from 'react'
import { Grid } from 'xn-reflex'

export default class PanelSection extends React.Component {
  render() {
    let { label, content, small, ...otherProps } = this.props

    let s = {}

    s.label = {
      color: '#A6A5B0',
      textTransform: 'uppercase',
      fontSize: '12px',
    }

    s.content = {
      color: '#FFFFFF',
      fontSize: small ? '24px' : '32px',
      lineHeight: '32px',
      fontFamily: 'Klartext Mono Thin',
    }

    return (
      <Grid style={{ flexDirection: 'column' }} {...otherProps}>
        <div style={s.label}>{label}</div>
        <div style={s.content}>{content}</div>
      </Grid>
    )
  }
}
```

And then 
```javascript
          <Grid marginBottom={[1]}>
            <PanelSection
              size={6}
              label={CAM_STR(camera.orientation)}
              content={statusMessage}
            />
            <PanelSection
                size={6}
                align="bottom"
                label={__('BT')}
                content={<Icon name="bluetooth" />}
            />
          </Grid>
```
![screen shot 2017-09-06 at 8 17 19 pm](https://user-images.githubusercontent.com/784852/30144494-690124b4-9340-11e7-9fd9-6657b4040a5a.png)

which gets me the bluetooth icon and `BT` text aligned right, using `align="bottom"` and `flexDirection: 'column'` on the container.

**Comment by garretto** [comment]
Actually now just doing this instead to use the `justify` keyword instead of `align`

```javascript
import React from 'react'
import { Grid } from 'xn-reflex'

export default class PanelSection extends React.Component {
  render() {
    let { label, content, small, justify, ...otherProps } = this.props

    let s = {}

    s.label = {
      color: '#A6A5B0',
      textTransform: 'uppercase',
      fontSize: '12px',
    }

    s.content = {
      color: '#FFFFFF',
      fontSize: small ? '24px' : '32px',
      lineHeight: '32px',
      fontFamily: 'Klartext Mono Thin',
    }

    return (
      <Grid style={{ flexDirection: 'column', textAlign: justify }} {...otherProps}>
        <div style={s.label}>{label}</div>
        <div style={s.content}>{content}</div>
      </Grid>
    )
  }
}
```

**Comment by obartra** [comment]
Hey there @garretto,

Sorry to hear you are having a hard time with the lib. I took a crack at implementing your demo without CSS here: https://codesandbox.io/s/53kqxw117p

Note that Reflex works well when you wrap your styleguide components with `<Grid>` but when you use it directly on divs it ends up being fairly verbose. That's why I created a couple components in that demo.

The downside of using custom flexbox properties is that you rely on the internal implementation of reflex which may not remain the same between minor/patch versions (although it is fairly stable now). You also loose the cross browser support testing that comes with pure reflex usage.

In general I think these are common problems when transitioning from flexbox, my recommendation is to try to resist the flexbox approach and embrace the full reflex power! 💥
